docs/DeveloperGuide.md

```diff
@@ -1,34 +1,509 @@
+</div>
+
+#### Design consideration
+
+The following options were considered when implementing commands:
+
+* Option 1 (Current Implementation): Implementing each command as a class by itself
+    * Pros: Increases modularity of code, higher overall code quality 
+    * Cons: More complicated to implement
+* Option 2: Implementing each command as a method in a class
+    * Pros: Easier to implement
+    * Cons: Class needs to be instantiated and increases coupling, reducing testability. This method also decreases SLAP.
+
+### Academic Calendar Planner: Edit Module Feature
+
+#### Current implementation
+
+Similar to the add module command, the edit module command is also executed by `AcademicPlannerParser`. It allows the user to edit the existing modules added to their `Academic Planner` by accessing the specified `PartialModule` object within the `userModuleList`and `userModuleMap`. 
+
+Given below is an example usage scenario and how add module command behaves at each step.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/editModuleCommand_initialState.png" alt="Initial state diagram for Edit Module Command"/>
+</div>
+
+
+__Step 1:__ The user calls the edit module command from the `AcademicPlannerParser` and  then `EditModuleCommand` will be initialized where its constructor would take in the same parameters as that of `AddModuleCommand`.
+
+__Step 2:__ The `execute()` method is called from the instance of `EditModuleCommand` which only throws `AcademicException` if applicable.
+
+__Step 3:__ Method `isModTakenByUser()` of the `ModuleValidator` is called to check if the `moduleCode` entered by the user exists within the `userModuleList` and `userModuleMap`.
+
+__Step 4:__ `in` reads the next line of input for user's choice of modifying either the semester or grade of the selected `moduleCode`.
+
+__Step 5:__ `isValidSemester()` or `isValidGrade()` is called to validate the semester or grade entered by the user.
+
+__Step 6:__ `updateModuleSemester()` or `updateModuleGrade()` is then called to conduct necessary changes to the information by accessing the module from `userModuleMap` and `userModuleList`.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/editModuleCommand_finalState.png" alt="Final state diagram for Edit Module Command"/>
+</div>
+
+
+__Step 7:__ `EditModuleCommand`, `EditUtils` and `ModuleValidator` are terminated.
+
+The following sequence diagram shows how `EditModuleCommand` works.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/editModuleCommand_sequence.png" alt="Sequence diagram for Edit Module Command"/>
+</div>
+
+
+
+The following diagram summarizes what happens when the user executes a `EditModuleCommand`: 
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/editModuleCommand_activity.png" alt="Activity diagram for Edit Module Command"/>
+</div>
```

Comment:
As not everyone will know how to read this diagram, perhaps add an elaborate explanation with the diagram. I think there are multiple occurences